common/common.go

```diff
@@ -653,3 +654,25 @@ func GetTypeAssertError(required string, received interface{}, fieldDescription
 	}
 	return fmt.Errorf("%w from %T to %s%s", ErrTypeAssertFailure, received, required, description)
 }
+
+// Batch takes a slice type and converts it into a slice of slices
+func Batch[S ~[]E, E any](blobs S, batchSize int) (batches []S) {
+	var j int
+	for i := 0; i < len(blobs); i += batchSize {
+		j += batchSize
+		if j >= len(blobs) {
+			j = len(blobs)
+		}
+		batches = append(batches, blobs[i:j])
+	}
+	return
```

Comment:
```suggestion
	if batchSize <= 1 {
		return
	}
	batches = make([]S, 0, (len(blobs)+batchSize-1)/batchSize)
	var j int
	for i := 0; i < len(blobs); i += batchSize {
		j += batchSize
		if j >= len(blobs) {
			j = len(blobs)
		}
		batches = append(batches, blobs[i:j])
	}
	return
```

Awesome function! Fixes panic and infinite loop issues and reduces allocations.

Reply:
batchSize 1 should return a new Clone of the slice, right?
So 660 should be `<= 0`

Reply:
Yeah good point. 🙏

Reply:
Fixed bd970c0

Can Haz Tests.